app/screens/terms_of_service/terms_of_service.js

```diff
@@ -0,0 +1,272 @@
+                }),
+                [{
+                    text: intl.formatMessage({id: 'mobile.terms_of_service.alert_ok', defaultMessage: 'Ok'}),
+                    onPress: async () => {
+                        await actions.logout();
+                        this.setNavigatorButtons(true);
+                        this.props.navigator.dismissAllModals();
+                    },
+                }],
+            );
+        }
+    };
+
+    handleAcceptTerms = async () => {
+        await this.registerUserAction(
+            true,
+            () => {
+                this.props.navigator.dismissModal({
+                    animationType: 'slide-down',
+                });
+            }
+        );
+    };
+
+    handleRejectTerms = async () => {
+        const {actions, siteName} = this.props;
+        const {intl} = this.context;
+
+        await this.registerUserAction(
+            false,
+            () => {
+                Alert.alert(
+                    this.props.siteName,
+                    intl.formatMessage({
+                        id: 'mobile.terms_of_service.terms_rejected',
+                        defaultMessage: 'You must agree to the terms of service before accessing {siteName}. Please contact your System Administrator for more details.',
+                    }, {
+                        siteName,
+                    }),
+                    [{
+                        text: intl.formatMessage({id: 'mobile.terms_of_service.alert_ok', defaultMessage: 'Ok'}),
+                        onPress: async () => {
+                            await actions.logout();
+                            this.setNavigatorButtons(true);
+                            this.props.navigator.dismissAllModals();
+                        },
+                    }],
+                );
+            }
+        );
+    };
+
+    registerUserAction = async (accepted, success) => {
+        const {actions} = this.props;
+        const {intl} = this.context;
+
+        this.setNavigatorButtons(false);
+        this.setState({
+            loading: true,
+        });
+
+        const {data} = await actions.updateTermsOfServiceStatus(this.state.termsId, accepted);
+        if (data) {
+            success(data);
+            this.setNavigatorButtons(true);
+            this.setState({
+                loading: false,
+            });
+        } else {
+            Alert.alert(
+                this.props.siteName,
+                intl.formatMessage({
+                    id: 'terms_of_service.api_error',
+                    defaultMessage: 'Unable to complete the request. If this issue persists, contact your System Administrator.',
+                }),
+                [{
+                    text: intl.formatMessage({id: 'mobile.terms_of_service.alert_ok', defaultMessage: 'OK'}),
+                    onPress: async () => {
+                        await actions.logout();
```

Comment:
Again why logout and not let the user retry?

Reply:
Added an option to retry as well